contracts/Aavegotchi/libraries/LibDelegatedWearables.sol

```diff
@@ -0,0 +1,85 @@
+        uint256 depositIdToUnequip = gotchiInfo.equippedDepositIds[_slot];
+
+        // remove wearable from Aavegotchi and delete delegation
+        ItemRolesInfo storage depositInfo = s.itemRolesDepositInfo[depositIdToUnequip];
+        bool _sameHandDelegationEquipped = (_slot == LibItems.WEARABLE_SLOT_HAND_LEFT && // if is hand left and right has the same delegation as this
+            gotchiInfo.equippedDepositIds[LibItems.WEARABLE_SLOT_HAND_RIGHT] == depositIdToUnequip) ||
+            // if is hand right and left has the same delegation as this
+            (_slot == LibItems.WEARABLE_SLOT_HAND_RIGHT && gotchiInfo.equippedDepositIds[LibItems.WEARABLE_SLOT_HAND_LEFT] == depositIdToUnequip);
+
+        // it will only skip, when the same hand delegation is equipped in the other hand
+        // to avoid removing the delegation from the depositInfo while the depositId still equipped in the other hand
+        if (!_sameHandDelegationEquipped) {
+            depositInfo.equippedGotchis.remove(_gotchiId);
+        }
+
+        depositInfo.balanceUsed--;
+        // this counter is decremented whenever a delegated wearable is unequipped
+        // this is used to allow Aavegotchis to be transferred or listed again (when the counter reaches 0)
+        gotchiInfo.equippedDelegatedWearablesCount--;
+        delete gotchiInfo.equippedDepositIds[_slot];
+    }
+
+    /**
+     * @notice Adds a delegated wearable to an Aavegotchi
+     * @dev Transfers the wearable from the diamond to the Aavegotchi and updates delegation info
+     * @param _depositIdToEquip The depositId of the Wearable to be equipped
+     * @param _gotchiId The id of the Aavegotchi
+     * @param _wearableIdToEquip The id of the wearable to be equipped
+     */
+    function addDelegatedWearableToGotchi(uint256 _depositIdToEquip, uint256 _gotchiId, uint256 _wearableIdToEquip) internal {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+        GotchiEquippedDepositsInfo storage _gotchiInfo = s.gotchiEquippedDepositsInfo[_gotchiId];
+        ItemRolesInfo storage depositInfo = s.itemRolesDepositInfo[_depositIdToEquip];
+
+        require(depositInfo.roleAssignment.grantee == LibMeta.msgSender(), "ItemsFacet: Wearable not delegated to sender or depositId not valid");
+        require(depositInfo.roleAssignment.expirationDate > block.timestamp, "ItemsFacet: Wearable delegation expired");
+        require(
+            s.itemRolesDepositInfo[_depositIdToEquip].deposit.tokenId == _wearableIdToEquip,
+            "ItemsFacet: Delegated Wearable not of this delegation"
+        );
+        require((depositInfo.deposit.tokenAmount - depositInfo.balanceUsed) > 0, "ItemsFacet: Not enough delegated balance");
+
+        // this counter is incremented whenever a delegated wearable is equipped
+        // this is used to prevent Aavegotchis equipped with delegated Wearables from being transferred or listed
+        _gotchiInfo.equippedDelegatedWearablesCount++;
```

Comment:
I wonder if would be cheaper to use a bool here instead of a counter? And only update the bool state from `true` to `false` when the last delegated wearable is unequipped? @karacurt

Reply:
The counter is necessary to keep track of the number of delegated Wearables an Aavegotchi has equipped. When `_gotchiInfo.equippedDelegatedWearablesCount == 0` the Aavegotchi is transferrable, when `_gotchiInfo.equippedDelegatedWearablesCount > 0` the Aavegotchi is not. If we have five delegated Wearables equipped, and the owner removes three, we are left with two, and the Aavegotchi remains not transferable. 

Removing the counter will make a code that is already complex even more complicated (we have to loop through the Wearables slots to count the number of delegated Wearables every time). This change will also likely make the `transferFrom` function more expensive